Utils/Elasticsearch/Oracle2JSON.py

```diff
@@ -119,6 +102,14 @@ def squash_records(rec):
         grouper[d['taskid']][d['type']].append(d['datasetname'])
     return [dict(taskid=k, **v) for k, v in grouper.items()]
 
+
+def join_results(tasks, datasets):
+    d = defaultdict(dict)
+    for l in (tasks, datasets):
+        for elem in l:
+            d[elem['taskid']].update(elem)
+            sys.stdout.write(json.dumps(str(d[elem['taskid']])) + '\n')
```

Comment:
First we will have all the elements of the `datasets` in the output, and only then -- the joined records.
The following stage will skip lines without necessary information (like `campaign` or `hashtag_list`), so it is OK for now, but to fix, we could output data in a loop over `d` after the join loop instead of doing it within.